airflow/ui/src/queries/useLogs.tsx

```diff
@@ -49,8 +50,46 @@ const parseLogs = ({ data }: ParseLogsProps) => {
     return { data, warning };
   }
 
-  // eslint-disable-next-line react/no-array-index-key
-  const parsedLines = lines.map((line, index) => <p key={index}>{line}</p>);
+  let startGroup = false;
+  let groupLines: Array<string> = [];
+  let groupName = "";
+
+  /* eslint-disable react/no-array-index-key */
+  const parsedLines = lines.map((line, index) => {
```

Comment:
This is a logic block which would be perfect to a unit test. Could you extract this into a functions and add a test case? (I know we have a very low coverage but this is an ideal candidate)

Reply:
Thanks, added tests along with testing click to expand of the section,